source/data-modeling/associations.txt

```diff
@@ -0,0 +1,504 @@
+
+Embedded matching on loaded documents has the following known limitations:
+
+- Embedded matching is not implemented for the following:
+  
+  - :ref:`Text search <mongoid-text-search>`
+  - :manual:`Geospatial query operators </reference/operator/query-geospatial/>`
+  - Operators that execute JavaScript code, such as :manual:`$where </reference/operator/query/where/>`
+  - operators that are implemented through other server functionality, such as
+    :manual:`$expr <https://mongodb.com/docs/manual/reference/operator/query/expr/>`
+    and :manual:`$jsonSchema
+    <https://mongodb.com/docs/manual/reference/operator/query/jsonSchema/>`
+
+- {+odm+} expands ``Range`` arguments to hashes with ``$gte`` and ``$lte``
+  conditions. This can lead to invalid queries in some cases and raises a an
+  ``InvalidQuery`` exception.
+
+- With the ``$regex`` operator, you cannot specify a regular expression object
+  as a pattern while also providing options to the ``$options`` field. You can
+  only provide options if the regular expression pattern is a string.
+
+- MongoDB Server versions 4.0 and earlier do not strictly validate ``$type``
+  arguments.
+
+Omit _id Fields
+~~~~~~~~~~~~~~~
+
+By default, {+odm+} adds an ``_id`` field to embedded documents. You can omit
+this field from embedded documents by explicitly specifying the ``_id`` field in
+your model and omitting the default value. The following example instructs {+odm+}
+not to add an ``_id`` field to the ``Albums`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embedded-omit-id
+   :end-before: # end-embedded-omit-id
+   :emphasize-lines: 4
+
+In the preceding ``Albums`` class, the ``_id`` field is not automatically added.
+Without a default value, {+odm+} does not store the value in the database
+unless it is provided.
+
+Delete Embedded Associations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can delete child documents from ``embeds_many`` associations by using one of
+the following methods:
+
+- ``clear()``
+- ``delete_all()``
+- ``destroy_all()``
+
+The ``clear()`` method uses the :manual:`$unset operator
+</reference/operator/update/unset>` operator to remove an entire embedded
+association from the parent document. The ``clear()`` method does not run any
+``destroy`` callbacks. The following example uses the ``clear()``
+method to remove all embedded associations from the ``Band`` class.
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-embedded-clear
+   :end-before: # end-embedded-clear
+
+The ``delete_all()`` method uses the :manual:`$pullAll operator
+</reference/operator/update/pullAll>` operator to remove documents in an
+embedded association. ``delete_all()`` loads the association if it has not
+yet been loaded, then only removes the documents that exist in the application.
+The ``delete_all()`` method does not run any ``destroy`` callbacks. 
```

Comment:
Q: would the reader most likely know what "destroy callbacks" are?

Reply:
I wondered that as well and looked into it, it seems like the destroy callbacks are things the user would define themselves on a model if they wanted them, so I think it's a Ruby practice that the user should know about.